code/src/hardware_usart.cpp

```diff
@@ -1,65 +1,53 @@
 #include <hardware_usart.hpp>
-#include <uart_ports.hpp>
-namespace r2d2 {
 
-
-    bool hardware_usart_c::transmit_ready() {
-        return (hardware_usart->US_CSR & 2);
-    }
-
-    void hardware_usart_c::send_byte(const uint8_t &b) {
-        while (!transmit_ready()) {
+namespace r2d2::usart {
+    enum class peripheral {
+        peripheral_a,
+        peripheral_b
+    };
+
+    struct hw_usart_s {
+        Usart *usart;
+        uint32_t rx_mask;
+        uint32_t tx_mask;
+        Pio * pio;
+        peripheral periph;
+        uint16_t id;
+    };
+
+    const static hw_usart_s usart[uint8_t(usart_ports_c::UART_SIZE)] = {
```

Comment:
Constexpr

Reply:
Unfortunately in the array a reinterpret_cast is used so it doesnt compile with constexpr